.github/workflows/pr.yml

```diff
@@ -6,53 +6,111 @@ jobs:
+          node-version: 20.17.0
+      - name: Create keypair
+        run: solana-keygen new --no-bip39-passphrase
+        shell: bash
+      - name: Make Anchor.toml compatible with runner
+        run: "sed -i 's:/user/:/runner/:' Anchor.toml"
+        shell: bash
+      - name: Install Cargo toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rustc
       - name: Install packages
         run: yarn install --frozen-lockfile
-      - name: Lint
-        run: yarn lint
+      - name: Lint js
+        shell: bash
+        run: yarn lint-js
+      - name: Lint solidity
+        shell: bash
+        run: yarn lint-solidity
+      - name: Lint rust
+        shell: bash
+        run: yarn lint-rust
   test:
     name: Test
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [16]
+        node:
+          - 20
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
-      - name: Use Node ${{ matrix.node }}
+      - name: "Use Node ${{ matrix.node }}"
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node }}
-          cache: "yarn"
+          node-version: "${{ matrix.node }}"
+          cache: yarn
+      - name: Setup Anchor & Build
+        uses: metadaoproject/setup-anchor@v2
+        with:
+          anchor-version: 0.30.1
+          solana-cli-version: 1.18.21
+          node-version: 20.17.0
+      - name: Create keypair
+        run: solana-keygen new --no-bip39-passphrase
+        shell: bash
+      - name: Make Anchor.toml compatible with runner
+        run: "sed -i 's:/user/:/runner/:' Anchor.toml"
+        shell: bash
+      - name: Install Cargo toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rustc
+      - name: Cache Cargo dependencies
+        uses: Swatinem/rust-cache@v2
       - name: Install packages
         run: yarn install --frozen-lockfile
       - name: Build
         run: yarn build
-      - name: Test
-        run: yarn test
+      - name: Test evm-hardhat
+        shell: bash
+        run: yarn test-evm
+      - name: Test svm
+        shell: bash
+        run: yarn test-svm
   forge:
     name: Forge
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node: [16]
+        node:
+          - 20
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
-      - name: Use Node ${{ matrix.node }}
+      - name: "Use Node ${{ matrix.node }}"
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node }}
-          cache: "yarn"
+          node-version: "${{ matrix.node }}"
+          cache: yarn
+      - name: Setup Anchor & Build
+        uses: metadaoproject/setup-anchor@v2
+        with:
+          anchor-version: 0.30.1
+          solana-cli-version: 1.18.21
+          node-version: 20.17.0
```

Comment:
Will this break if the node version varies (i.e. 20 resolves to 20.17.1)?

Can we just specify 20 here or does it have to be specific?

Reply:
I think it will work just fine? the version specified above in the config has `20` set, with no sub versions. I'll try changing this here and see if the workflow still runs.

Reply:
test worked just fine with the node version simply being "20"

Reply:
Perfect